apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -0,0 +1,367 @@
+            : this(SpannerCommandTextBuilder.FromCommandText(commandText), connection, transaction, parameters)
+        {
+        }
+
+        /// <inheritdoc />
+        public override string CommandText
+        {
+            get { return SpannerCommandTextBuilder?.ToString() ?? string.Empty; }
+            set { SpannerCommandTextBuilder = SpannerCommandTextBuilder.FromCommandText(value); }
+        }
+
+        /// <inheritdoc />
+        public override int CommandTimeout
+        {
+            get { return _commandTimeout; }
+            set { _commandTimeout = value; }
+        }
+
+        /// <inheritdoc />
+        public override CommandType CommandType
+        {
+            get { return CommandType.Text; }
+            set
+            {
+                if (!Equals(value, CommandType.Text))
+                    throw new NotSupportedException("Cloud Spanner only supports CommandType.Text.");
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool DesignTimeVisible { get; set; }
+
+        /// <summary>
+        /// </summary>
+        public new SpannerParameterCollection Parameters { get; private set; }
+
+        /// <summary>
+        /// </summary>
+        public SpannerConnection SpannerConnection { get; set; }
+
+        /// <inheritdoc />
+        public override UpdateRowSource UpdatedRowSource
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <inheritdoc />
+        protected override DbConnection DbConnection
+        {
+            get { return SpannerConnection; }
+            set { SpannerConnection = (SpannerConnection) value; }
+        }
+
+        /// <inheritdoc />
+        protected override DbParameterCollection DbParameterCollection => Parameters;
+
+        /// <inheritdoc />
+        protected override DbTransaction DbTransaction
+        {
+            get { return _transaction; }
+            set { _transaction = (SpannerTransaction) value; }
+        }
+
+        /// <summary>
+        /// </summary>
+        internal SpannerCommandTextBuilder SpannerCommandTextBuilder { get; set; }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        public object Clone()
+        {
+            return new SpannerCommand {
+                DesignTimeVisible = DesignTimeVisible,
+                Parameters = Parameters,
+                SpannerConnection = SpannerConnection,
+                SpannerCommandTextBuilder = SpannerCommandTextBuilder,
+                CommandTimeout = CommandTimeout
+            };
+        }
+
+        /// <inheritdoc />
+        public override void Cancel()
+        {
+            _synchronousCancellationTokenSource.Cancel();
+        }
+
+        /// <inheritdoc />
+        public override int ExecuteNonQuery()
+        {
+            return ExecuteNonQueryAsync(_synchronousCancellationTokenSource.Token).Result;
```

Comment:
Added ResultWithUnwrapped.  re: is it safe?  I reviewed all called awaits and indeed I missed one ConfigureAwait(false).
However, I do not have a better alternative and in fact you can make the argument that the developer can just as easily call ".Result" on our async methods and expect them not to deadlock.
So the burden is really on making the async calls safe -- which should be a set of test cases (every public async method can be called with Result on the main thread).